CS483/CS483/Kartaclysm/StateMachine/Gameplay/StateRacing.cpp

```diff
@@ -267,9 +280,43 @@ void Kartaclysm::StateRacing::PauseGame(const HeatStroke::Event* p_pEvent)
 	m_pStateMachine->Push(STATE_PAUSED, mContext);
 }
 
+void Kartaclysm::StateRacing::RacerFinishedRace(const HeatStroke::Event* p_pEvent)
+{
+	std::string strRacerId = "";
+	p_pEvent->GetRequiredStringParameter("racerId", strRacerId);
+
+	float fRacerTime = 0.0f;
+	p_pEvent->GetRequiredFloatParameter("racerTime", fRacerTime);
+
+	m_vRaceResults.push_back({strRacerId, fRacerTime});
+
+	if (m_vRaceResults.size() >= m_uiNumRacers)
```

Comment:
I'm guessing this is a catch all, as I'm not sure when the results size should be bigger than the number of racers.